### PR TITLE
Wire up individual item deletion/download icons

### LINKED ISSUE #60: Hook up individual item delete and download buttons
This refers to providing the correct behavior implementation for the new UI elements added in #56.

### PR DESCRIPTION
**Pull request by waxlamp**
This PR provides action semantics for the item-level download and delete icons, for workspaces, networks, and tables.

To accomplish this more easily, I decided to respecialize the `ItemPanel` component into `TablePanel` and `NetworkPanel` components. Unfortunately, there are some (reasonable) limitations in Vue around accessing slot content via refs, and while using refs to manipulate child components like `v-dialog` instances is not a best practice, it seemed to be warranted in this case.

In the future, we may want to do the following:
- use actual best practices to handle the lifecycle of the various dialogs in use in these panels
- refactor the repeated logic between the two panels into a mixin

But I don't think of these as at all urgent. If we need to make changes to these panels, or add a third panel, then these may become more important.

For now, please mess around with these and see if I missed any corner cases in behavior. If these are good, I want to get this merged and conclude our usability sprint :tada:

Closes #60

### CODE REVIEW

**Comment by JackWilb** [comment]
Looks to work pretty well, I'll dig into the code a little later. I noticed one thing so far:

When hitting delete on any table that is used in a network, the dialog pops up and then immediately changes to the "you can't delete that yet" dialog. Is there a way we can have it pop up with the right message right away?

**Comment by waxlamp** [comment]
> When hitting delete on any table that is used in a network, the dialog pops up and then immediately changes to the "you can't delete that yet" dialog. Is there a way we can have it pop up with the right message right away?

Good catch. I am noticing this behavior on multinet.app as well, so I've opened #67 to address it, but it shouldn't block the review/merge of this PR.

**Comment by waxlamp** [comment]
> Works well locally, requested a small change. Somes notes:
> 
> * When [vuex](https://github.com/multinet-app/multinet-client/issues/37) is added, it will clean up this portion of code a bit, because the list of networks/tables will be managed by the store.
> 

For sure.

> * Looking at what's necessary for this, I think `DeleteWorkspaceDialog` should contain just the dialog, and not the activator button or anything. This would make more sense as far as the structure, and would allow us to directly control the state of the dialog, and remove the need for the `singleSelected` variable.
> 

Let's definitely revisit this after the big VueX change lands. I might be missing something because even if we extract the activator icon from inside the dialog component (a good change in general, IMO), we would still need the singleSelected mechanism, or something like it, in order to properly keep track of both the checkbox state, and handle one-off delete/download actions.

> 
> I think all of this is for separate PRs though, so I'm okay with approving this once my comments are addressed.

:confetti_ball:

File: src/components/TablePanel.vue [hunk 1]
```diff
@@ -59,93 +68,137 @@
...
     },
+
+    cleanup(deleted?: string[]) {
+      const {
+        singleSelected,
+      } = this;
+
+      if (singleSelected !== null) {
+        this.singleSelected = null;
+      }
```
Comment (JackWilb) [inline src/components/TablePanel.vue:194]: Can you simplify this to just: `this.singleSelected = null`?

File: src/components/NetworkPanel.vue [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+    },
+
+    cleanup(deleted?: string[]) {
+      const {
+        singleSelected,
+      } = this;
+
+      if (singleSelected !== null) {
+        this.singleSelected = null;
+      }
```
Comment (JackWilb) [inline src/components/NetworkPanel.vue:205]: Same as the other.

File: src/components/TablePanel.vue (hunk 1 above)
Comment (waxlamp) [reply src/components/TablePanel.vue]: See 14dcbf1dbaf16da19aa4551782d5a6419fad6314.

File: src/components/NetworkPanel.vue (hunk 2 above)
Comment (waxlamp) [reply src/components/NetworkPanel.vue]: See 14dcbf1dbaf16da19aa4551782d5a6419fad6314.

File: src/components/TablePanel.vue (hunk 1 above)
Comment (JackWilb) [reply src/components/TablePanel.vue]: Not a huge deal, but I think you can also delete the 
```
     const {
        singleSelected,
      } = this;
```

Comment (waxlamp) [reply src/components/TablePanel.vue]: Good catch. See f25502a.

**Pull request merged**